test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionTest.scala

```diff
@@ -226,13 +226,100 @@ trait CollectionTest {
     assertEquals(5, coll.size())
     assertIteratorSameElementsAsSet(-45, 0, 12, 32, 42)(coll.iterator())
   }
+
+  @Test def toStringShouldConvertEmptyCollection(): Unit = {
+    val coll = factory.empty[Double]
+
+    val expected = "[]"
+    val result = coll.toString()
+    assertEquals(expected, result)
+  }
+
+  @Test def toStringShouldConvertOneElementCollection(): Unit = {
+    val coll = factory.fromElements[Double](1.01)
+
+    // Javascript displays n.0 as n, so one trailing digit must be non-zero.
+    val expected = "[1.01]"
+    val result = coll.toString()
+    assertEquals(expected, result)
+  }
+
+  @Test def toStringShouldUseCommaSpace(): Unit = {
+
+     // Choose Doubles which display the same in Java and Scala-js.
+    // Javascript displays n.0 as n, so one trailing digit must be non-zero.
+    val elements = Seq(88.42, -23.36, 60.173)
+
+    val coll = factory.fromElements[Double](elements:_*)
+
+    // The order of elements returned by each collection is defined
+    // by the collection. Be prepared to handle the general case of any
+    // order here. Specific collections should test the order they specify.
+    val answers = elements.permutations.map(_.mkString("[", ", ", "]"))
+
+    val expected = answers.toSet
+
+    val result = coll.toString()
+
+    val resultIsValid =  expected.contains(result)
+
+    val msg = "result '" + result +
+        "' not in expected set '" + expected + "'"
+
+    assertTrue(msg, resultIsValid)
+  }
+
+  @Test def toStringShouldHandleNullElements(): Unit = {
+    if (factory.allowsNullElement) {
+       val elements = Seq(-1, -2, null, -3)
+
+       val coll = factory.fromElements[Any](elements:_*)
+
+       val answers = elements.permutations.map(_.mkString("[", ", ", "]"))
+       val expected = answers.toSet
+
+       val result = coll.toString()
+       val resultIsValid =  expected.contains(result)
+
+       val msg = "result '" + result +
+           "' not in expected set '" + expected + "'"
+
+    assertTrue(msg, resultIsValid)
+    }
+  }
+
+  @Test def toStringInCustomClassShouldWork(): Unit = {
+
+    case class Custom(name: String, id: Int) extends Ordered[Custom] {
+      def compare(that: Custom) = this.id - that.id
```

Comment:
This method is public, so it should have an explicit result type. (This is supposed to be enforced by Scalastyle; I don't why it didn't report it here, maybe because it is a local class.)

Reply:
Fixed.